packages/@ember/-internals/container/lib/registry.ts

```diff
@@ -468,116 +447,32 @@ export default class Registry implements IRegistry {
-
-   user.router instanceof Router; //=> true
-   post.router instanceof Router; //=> true
-
-   // both controllers share the same router
-   user.router === post.router; //=> true
-   ```
-
-   @private
-   @method typeInjection
-   @param {String} type
-   @param {String} property
-   @param {String} fullName
-   */
-  typeInjection(type: string, property: string, fullName: string): void {
-    assert('fullName must be a proper full name', this.isValidFullName(fullName));
-
-    let fullNameType = fullName.split(':')[0];
-    assert(`Cannot inject a '${fullName}' on other ${type}(s).`, fullNameType !== type);
-
-    let injections = this._typeInjections[type] || (this._typeInjections[type] = []);
-
-    injections.push({ property, specifier: fullName });
-  }
-
-  /**
-   Defines injection rules.
-
-   These rules are used to inject dependencies onto objects when they
-   are instantiated.
-
-   Two forms of injections are possible:
+   This is deprecated in favor of explicit injection of dependencies.
 
-   * Injecting one fullName on another fullName
-   * Injecting one fullName on a type
-
-   Example:
-
-   ```javascript
-   let registry = new Registry();
-   let container = registry.container();
-
-   registry.register('source:main', Source);
-   registry.register('model:user', User);
-   registry.register('model:post', Post);
-
-   // injecting one fullName on another fullName
-   // eg. each user model gets a post model
-   registry.injection('model:user', 'post', 'model:post');
-
-   // injecting one fullName on another type
-   registry.injection('model', 'source', 'source:main');
-
-   let user = container.lookup('model:user');
-   let post = container.lookup('model:post');
-
-   user.source instanceof Source; //=> true
-   post.source instanceof Source; //=> true
-
-   user.post instanceof Post; //=> true
-
-   // and both models share the same source
-   user.source === post.source; //=> true
+   Reference: https://deprecations.emberjs.com/v3.x#toc_implicit-injections
    ```
 
    @private
    @method injection
    @param {String} factoryName
    @param {String} property
    @param {String} injectionName
+   @deprecated
    */
-  injection(fullName: string, property: string, injectionName: string) {
-    assert(
-      `Invalid injectionName, expected: 'type:name' got: ${injectionName}`,
-      this.isValidFullName(injectionName)
+  injection(fullName: string, property: string) {
+    deprecate(
+      `As of Ember 4.0.0, owner.inject no longer injects values into resolved instances, and calling the method has been deprecated. Since this method no longer does anything, it is fully safe to remove this injection. As an alternative to this API, you can refactor to explicitly inject \`${property}\` on \`${fullName}\`, or look it up directly using the \`getOwner\` API.`,
+      false,
+      {
+        id: 'remove-owner-inject',
+        until: '5.0.0',
+        url: 'https://deprecations.emberjs.com/v3.x#toc_implicit-injections',
```

Comment:
This should get a 4.x deprecation entry. The 3.x one is much more information than is probably needed, explains how to move away from APIs which the user already shouldn't be using, and is just confusing as it is in the 3.x series.

![Screenshot from 2021-08-19 21-36-13](https://user-images.githubusercontent.com/8752/130165363-57d1148b-5e28-42a7-9dd0-598049352387.png)

Reply:
https://github.com/ember-learn/deprecation-app/pull/932

Reply:
If you can update the link here to point to the correct 4.x deprecation page we can land this PR independent of https://github.com/ember-learn/deprecation-app/pull/932, IMO

Reply:
ok fixed!